### PR TITLE
Retry pantry requests

### PR DESCRIPTION
**Pull request by jmpesp**
Retry pantry requests in the face of network weather (if there is a timeout, 503, or 429).

Also, provide a Logger in PantryEntry, and log when a job fails.

### CODE REVIEW

**Comment by jmpesp** [comment]
Good point, see 08116b6

**Review by leftwo — APPROVED** [review]
I'm at 132 loops so far with these changes when running the
`test_pantry_import_from_url_ovmf` test.  Before these changes I 
only ever get to 25 loops, but usually failed before 10.

File: pantry/src/pantry.rs [hunk 1]
```diff
@@ -94,13 +94,22 @@ impl PantryEntry {
...
-        // validate the URL can be reached, and grab the content length
-        let dur = std::time::Duration::from_secs(5);
+        // Construct a reqwest client that
+        //
+        // 1) times out after 5 seconds if a connection can't be made
+        // 2) times out if the connection + chunk download takes over 60 seconds
+        //
+        // Now, `MAX_CHUNK_SIZE / 60s ~= 8.5kb/s`. If the connection you're downloading from is
+        // that slow, then the pantry won't work, sorry!
+        let connect_timeout = std::time::Duration::from_secs(5);
```
[the remark this answers is not in the document]
Comment (jmpesp) [reply pantry/src/pantry.rs:104]: I'll do 10, nbd. Whatever value we choose is wrong, remember? :)

**Comment by leftwo** [comment]
On my dogfood host, a timeout of 25 got me 10 loops before failure, confirming that any value will be wrong.

**Pull request merged**